edgedb/retry.py

```diff
@@ -0,0 +1,163 @@
+import asyncio
+import random
+import time
+
+from . import errors
+from . import transaction as _transaction
+
+
+DEFAULT_MAX_ITERATIONS = 3
+
+
+def default_backoff(attempt):
+    return (2 ** attempt) * 0.1 + random.randrange(100) * 0.1
+
+
+class AsyncIOIteration(_transaction.AsyncIOTransaction):
+    def __init__(self, retry, owner):
+        super().__init__(owner)
+        self.__retry = retry
+
+    async def start(self):
+        if not self._managed:
+            raise errors.InterfaceError(
+                "Only managed retriable transactions are supported. "
+                "Use `async with transaction:`"
+            )
+        # TODO(tailhook) if this is not the first iteration suppress
+        # `wait_until_available` timeout
```

Comment:
You mean if the connection broke on the second iteration we don't want to wait on connect?

Reply:
If so let's implement before merging this?

Reply:
Done.